### ISSUE TITLE
Gecko and Gaia forks being out-of-sync sometimes breaks Gaia tests in Marionette

### ISSUE THREAD

**Issue by jonallengriffin** [andreasgal/B2G#171]
Gaia apps, and therefore tests of the Gaia apps, rely on both code from gecko and gaia.  Marionette runs the Gaia tests from the main B2G repo, but the gaia and gecko submodules here are often out-of-sync (e.g., right now the gecko submodule was updated 1 day ago, but the gaia submodule was updated 5 days ago).  This sometimes breaks the version of the Gaia tests that Marionette runs; sometimes code in one submodule will depend on code in the other, but they won't be updated together, and the result is that tests fail.

Sometimes Marionette itself receives updates to support the Gaia tests, which adds another layer to the submodule sync problem.

I'm not sure what  the solution here is, if any.  We plan on implementing an auto-merge system from m-c into the gecko fork; perhaps we could automatically update the gecko and gaia submodules at the same time (after MWC)?

**Comment by vingtetun** [comment]
On 09/02/2012 20:13, Jonathan Griffin wrote:

> Gaia apps, and therefore tests of the Gaia apps, rely on both code from gecko and gaia.  Marionette runs the Gaia tests from the main B2G repo, but the gaia and gecko submodules here are often out-of-sync (e.g., right now the gecko submodule was updated 1 day ago, but the gaia submodule was updated 5 days ago).  This sometimes breaks the version of the Gaia tests that Marionette runs; sometimes code in one submodule will depend on code in the other, but they won't be updated together, and the result is that tests fail.
> 
> Sometimes Marionette itself receives updates to support the Gaia tests, which adds another layer to the submodule sync problem.
> 
> I'm not sure what  the solution here is, if any.  We plan on implementing an auto-merge system from m-c into the gecko fork; perhaps we could automatically update the gecko and gaia submodules at the same time (after MWC)?

I also would be more than happy to have a solution here. But sadly I 
have no good ideas to suggest.

**Comment by joneschrisg** [comment]
Yes, auto-merging gaia pushes should be quite doable.  Trigger a try-merge on pushes to either gaia or gecko.  If tests fail, and the try-merge was triggered by a gecko push, blame the gecko push.  If triggered by gaia push, blame the gaia patch.  If both changed, blame both.

**Comment by jonallengriffin** [comment]
Ok, we'll work on implementing that as soon as we get another EC2 VM.
